libsolidity/codegen/LValue.cpp

```diff
@@ -338,10 +338,30 @@ void StorageItem::storeValue(Type const& _sourceType, SourceLocation const& _loc
 		if (m_dataType->category() == Type::Category::Array)
 		{
 			m_context << Instruction::POP; // remove byte offset
-			ArrayUtils(m_context).copyArrayToStorage(
-				dynamic_cast<ArrayType const&>(*m_dataType),
-				dynamic_cast<ArrayType const&>(_sourceType)
-			);
+			auto const& sourceArrayType = dynamic_cast<ArrayType const&>(_sourceType);
+			if (sourceArrayType.baseType()->category() == Type::Category::Struct && sourceArrayType.location() != DataLocation::Storage)
+			{
+				bool fromCalldata = sourceArrayType.location() == DataLocation::CallData;
+				solAssert(_sourceType.sizeOnStack() == (fromCalldata ? 2 : 1), "");
```

Comment:
Only dynamically sized calldata arrays occupy two stack slots - for statically sized calldata arrays there's no length on the stack, so I think the static case is missing here, right?

Reply:
Yes, only dynamic arrays from calldata have 2 stack slots. Fixing now.